src/apps/chifra/pkg/config/block_scrape.go

```diff
@@ -0,0 +1,77 @@
+/*-------------------------------------------------------------------------------------------
+ * qblocks - fast, easily-accessible, fully-decentralized data from blockchains
+ * copyright (c) 2016, 2021 TrueBlocks, LLC (http://trueblocks.io)
+ *
+ * This program is free software: you may redistribute it and/or modify it under the terms
+ * of the GNU General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version. This program is
+ * distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details. You should have received a copy of the GNU General
+ * Public License along with this program. If not, see http://www.gnu.org/licenses/.
+ *-------------------------------------------------------------------------------------------*/
+package config
+
+import (
+	"runtime"
+	"strings"
+
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/logger"
+	"github.com/spf13/viper"
+)
+
+var blockScrapeViper = viper.New()
+var blockScrapeRead = false
+var cachedBlockScrape BlockScrape
+
+type blockScrapeRequires struct {
+	Tracing  bool
+	Parity   bool
+	Balances bool
+	Archive  bool
+}
+
+type blockScrapeDev struct {
+	IpfsGateway string `mapstructure:"ipfs_gateway"`
+	MaxPoolSize int
+}
+
+type blockScrapeUnchainedIndex struct {
+	Address              string
+	ManifestHashEncoding string
+}
+
+type BlockScrape struct {
+	Requires       blockScrapeRequires
+	Dev            blockScrapeDev
+	UnchainedIndex blockScrapeUnchainedIndex
+}
+
+// init sets up default values for the given configuration
+func init() {
```

Comment:
So, just for my information. Is it true that a `init` function in any file will get called prior to startup? In other words, prior to `main`, so we can count on these values being here before main?

Reply:
Yes, it's described here (5th paragraph from the end of the section): https://golang.org/ref/spec#Package_initialization